projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -54,30 +57,35 @@ static StatusCode prv_set_channel(Ads1015Storage *storage, Ads1015Channel channe
 // Also if there is a callback on a channel, it will be run here.
 static void prv_interrupt_handler(const GPIOAddress *address, void *context) {
   Ads1015Storage *storage = context;
+ gpio_it_mask_interrupt(&storage->ready_pin, true);
   Ads1015Channel current_channel = storage->current_channel;
   uint8_t channel_enable_bitset = storage->channel_enable_bitset;
   uint8_t read_conv_register[2] = { 0, 0 };
-
+if ((storage->channel_enable_bitset && (1 << current_channel)) != 0){
```

Comment:
Why is this necessary?

Reply:
very necessary :)) && should be & tho. I fixed that.